src/App.tsx

```diff
@@ -1,24 +1,221 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import {
+  FC,
+  useEffect,
+  useState,
+  useMemo,
+  useCallback,
+} from 'react';
+import './App.scss';
 import { UserWarning } from './UserWarning';
+import {
+  getTodos,
+  removeTodo,
+  addTodoToServer,
+  changeTodo,
+} from './api/todos';
+import { TodosList } from './components/TodosList/TodosList';
+import { Todo } from './types/Todo';
+import { ErrorInfo } from './components/ErrorInfo/ErrorInfo';
+import {
+  preparedTodos,
+  getCompletedTodoIds,
+  filteredTodosByCompletion,
+} from './utils/todoUtils';
+import { StatusValue } from './types/StatusValue';
+import { TodoAppHeader } from './components/TodoAppHeader/TodoAppHeader';
+import { TodoAppFooter } from './components/TodoAppFooter/TodoAppFooter';
 
-const USER_ID = 0;
+const USER_ID = 10725;
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [todoFilter, setTodoFilter] = useState<StatusValue>(StatusValue.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [visibleError, setVisibleError] = useState('');
+
+  const fetchTodosFromServer = async () => {
+    try {
+      const fetchedTodos = await getTodos(USER_ID);
+
+      setTodos(fetchedTodos as Todo[]);
+    } catch (error) {
+      if (error instanceof Error) {
+        setVisibleError(`Unable to load a todos: ${error.message}`);
+      }
+    }
+  };
+
+  useEffect(() => {
+    fetchTodosFromServer();
+  }, [tempTodo, loadingTodoIds]);
+
+  const completedTodoIds = useMemo(() => (
+    getCompletedTodoIds(todos)
+  ), [todos]);
+
+  const visibleTodos = useMemo(() => (
+    preparedTodos(todos, todoFilter)
+  ), [todos, todoFilter]);
+
+  const removeTodos = useCallback(async (todoIds: number[]) => {
```

Comment:
useCallback takes quite a few recources to memoise your function, you use it correctly in your app, but I suggest for the future use it not everywhere, but in cases where you see that you have problems with optimisation.

I will not reject because of it, but think about what i wrote ;)